pre-build/action.yml

```
@@ -67,7 +67,8 @@ runs:
 
     - name: Download bottles from GitHub Actions
       if: fromJson(inputs.download-bottles)
-      uses: actions/download-artifact@v3-node20
+      uses: actions/download-artifact@v4
       with:
-        name: bottles
+        pattern: bottles{,_*}
```

```suggestion
        pattern: bottles_*
```

Because v4 cannot download v3 artifacts anyway, according to https://github.com/actions/download-artifact/tree/v4/#v4---whats-new.

This affects dependent testing, which means some of the ongoing CI jobs will fail when this is merged. Preferably the long timeout jobs should be at dep testing (i.e., after bottles are downloaded) when merging this.

---

pr-pull uses REST API which wraps v3 and v4 that part is probably OK at least.

But yeah, perhaps time this for when the CI queue is empty. I can also watch out for that opportunity similarly if needed.